### PR TITLE
Stylish haskell after save

### PR DESCRIPTION
**Pull request by WraithM**
Okay! This actually works. It handles errors appropriately.

### CODE REVIEW

File: haskell-commands.el [hunk 1]
```diff
@@ -806,31 +806,21 @@ inferior GHCi process."
...
+         (err-buf (generate-new-buffer " error"))
          (default-directory (if (and (boundp 'haskell-session)
                                      haskell-session)
                                 (haskell-session-cabal-dir haskell-session)
                               default-directory))
-         (_errcode (with-temp-file tmp-file
-		     (call-process cmd filename
-				   (list (current-buffer) err-file) nil)))
+         (_errcode
+            (call-process-region (point-min) (point-max) cmd nil (list (buffer-name tmp-buf) (buffer-name err-buf)) nil))
```
Comment (gracjan) [inline haskell-commands.el]: AFAIK can't have buffer for STDERR, has to be a file.

Comment (WraithM) [inline haskell-commands.el]: This worked as expected. I tested it. I could do a few more tests or something to be absolutely certain.

File: haskell-commands.el [hunk 2]
```diff
@@ -806,31 +806,21 @@ inferior GHCi process."
...
-                    (setq str (replace-match "" t t str)))
-                  str))
-         (_errout (lambda (fmt &rest args)
-		    (let* ((warning-fill-prefix "    "))
-		      (display-warning cmd (apply 'format fmt args) :warning))))
-         (filename (buffer-file-name (current-buffer)))
-         (cmd-prefix (replace-regexp-in-string " .*" "" cmd))
-         (tmp-file (make-temp-file cmd-prefix))
-         (err-file (make-temp-file cmd-prefix))
+  (let* ((tmp-buf (generate-new-buffer " output"))
```
Comment (gracjan) [inline haskell-commands.el]: Note that this has to be protected by `unwind-protect`.

File: haskell-commands.el (hunk 1 above)
Comment (gracjan) [inline haskell-commands.el]: Interesting. It should be more like:

```elisp
(call-process-region (point-min) (point-max) cmd nil
          (list tmp-buf (make-temp-name "tempxx")) nil))
```

Comment (gracjan) [inline haskell-commands.el]: And again, temp file has to be deleted using `unwind-protect`.

**Comment by gracjan** [comment]
Can you rebase on top of current master?

**Comment by WraithM** [comment]
wasamasa on freenode #emacs helped me out a bit: https://gist.github.com/wasamasa/b89a54f56f3ef86d20cd6b8d85e2a5d8/961428e28279cb38955640211c6cbe3b3cc0fa6b

**Comment by gracjan** [comment]
Need to use `unwind-protect`. Need to use `insert-buffer-substring`.

**Comment by gracjan** [comment]
@WraithM: This is a lovely contribution, I would love to get this in. Can you rebease on mater and add `unwind-protect` where it is needed?

**Pull request closed without merging**